### PR TITLE
Feature/star rating bar

### PR DESCRIPTION
**Pull request by GreysonElkins**
#### What's this PR do?
* Adds a rating component for users to post and view ratings
#### Where should the reviewer start?
`Ratings.js` 
#### How should this be manually tested?
Needs Jest testing
#### Any background context you want to provide?
undefined states and default arguments are _**BOSS**_
#### What are the relevant tickets?
#42 #43 #9 
#### Questions:
Why isn't Birdman on here?

### CODE REVIEW

File: src/App/App.js [hunk 1]
```diff
@@ -34,23 +36,41 @@ class App extends Component {
...
   try {
-   const movies = await API.getData('movies');
-   this.setState({ movies });
-   this.setState({ pageView: "Home", error: '' });
+    const movies = await API.getData('movies')
+    if(this.state.isLoggedIn === true) {
+      API.getData(`users/${this.state.user.id}/ratings`)
+        .then(ratings => {
+        this.convertRatingsToStarValues(ratings)
+        this.setState({
```
Comment (leighlars) [inline src/App/App.js:44]: App is getting so many states on it. I wonder if this is typical for a real life app?

File: src/App/App.js [hunk 2]
```diff
@@ -120,23 +140,56 @@ class App extends Component {
...
+        error: "",
+      })})
   } else {
    this.setState({
     error: "Incorrect email or password. Please try again.",
    });
   }
  };
 
+ convertRatingsToStarValues = (ratings) => {
```
Comment (leighlars) [inline src/App/App.js:161]: Good catch on the division since we are doing 5 stars

File: src/App/App.js [hunk 3]
```diff
@@ -120,23 +140,56 @@ class App extends Component {
...
+  ratings.forEach((rating) => rating.rating = rating.rating / 2)
+ }
+
  logout = () => {
   this.setState({ pageView: "Home", isLoggedIn: false, user: "" });
  };
 
+ rateMovie = async (rating) => {
```
Comment (leighlars) [inline src/App/App.js:169]: So freaking good. Good job.

File: src/MovieCard/MovieCard.scss [hunk 4]
```diff
@@ -7,9 +7,8 @@
   margin: 1.5em;
   width: 15em;
   
-  img {
+  .poster {
```
Comment (leighlars) [inline src/MovieCard/MovieCard.scss:10]: Like the renaming here

File: src/MoviePage/MoviePage.js [hunk 5]
```diff
@@ -3,30 +3,51 @@ import Error from '../Error/Error'
...
      <button className="movie-trailer-btn">Play Trailer</button>
      <article className="movie-information">
-      <p className='movie-overview'><b>Overview:</b> {movie.overview}</p>
-      <p className='movie-genre'><b>Genre(s):</b> {movie.genres.join(", ")}</p>
-      <p className='movie-release-date'><b>Release Date:</b> {moment(movie.release_date).format("MMMM DD, YYYY")}</p>
-      <p className='movie-budget'><b>Budget:</b> ${movie.budget}</p>
-      <p className='movie-revenue'><b>Revenue:</b> ${movie.revenue}</p>
-      <p className='movie-runtime'><b>Runtime:</b> {movie.runtime} minutes</p>
-      <p className='movie-tagline'><b>Tagline:</b> {movie.tagline ? movie.tagline : "None"}</p>
+      <p className="movie-overview">
```
Comment (leighlars) [inline src/MoviePage/MoviePage.js:28]: Love the line breaks!!! You and your poetry background

File: src/Rating/Rating.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      if (i < numberOfStars) starSource = selectedStar
+      stars.push(
+        <div className="stars-box">
+          <img
+            // starValue={props.starValue}
+            className="star-image"
+            alt="empty star icon"
+            src={starSource}
+            id={i + 1}
+            onMouseEnter={() => this.setState({ currentValue: i + 1 })}
```
Comment (leighlars) [inline src/Rating/Rating.js:27]: Love that we went with the stars idea and using the mouse events to set the value.

File: src/MovieCard/MovieCard.scss (hunk 4 above)
Comment (GreysonElkins) [reply src/MovieCard/MovieCard.scss:10]: Lol I think u did that

File: src/MoviePage/MoviePage.js (hunk 5 above)
Comment (GreysonElkins) [reply src/MoviePage/MoviePage.js:28]: Haha again, I kind of think that was you!

File: src/Rating/Rating.js (hunk 6 above)
Comment (GreysonElkins) [reply src/Rating/Rating.js:27]: Me too but I don't know if I can handle going for half-stars after today

**Pull request merged**